### PR TITLE
Cost Model Error Reporting

### PR DESCRIPTION
**Pull request by mbolt35**

### CODE REVIEW

File: pkg/errors/panic.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+// PanicHandler is a func that receives a Panic and returns a bool representing whether or not
+// the panic should recover or not.
+type PanicHandler = func(p Panic) bool
+
+var (
+	enabled    = false
+	dispatcher = make(chan Panic)
+)
+
+// SetPanicHandler sets the handler that is executed when
```
Comment (nikovacevic) [inline pkg/errors/panic.go:46]: cliffhanger

File: pkg/costmodel/cluster.go [hunk 2]
```diff
@@ -40,7 +41,7 @@ const (
 // TODO move this to a package-accessible helper
 type PromQueryContext struct {
 	Client         prometheus.Client
-	ErrorCollector *util.ErrorCollector
+	ErrorCollector *errors.ErrorCollector
```
Comment (nikovacevic) [inline pkg/costmodel/cluster.go:44]: Complete nit, but might be nice if this is just called `errors.Collector` to avoid naming stutter.

**Comment by nikovacevic** [comment]
@mbolt35 a nice artifact for this PR might be a quick summary of how this was tested

File: pkg/errors/panic.go (hunk 1 above)
Comment (mbolt35) [reply pkg/errors/panic.go:46]: Yikes! Thanks :)

**Pull request merged**